spec/parsers/webp_parser_spec.rb

```diff
@@ -0,0 +1,66 @@
+require 'spec_helper'
+
+describe FormatParser::WebpParser do
+  it 'does not parse files with an invalid RIFF header' do
+    result = subject.call(File.open(fixtures_dir + 'WEBP/invalid-header.webp', 'rb'))
+    check_result(result, false)
+  end
+
+  it 'does not parse files with an unrecognised variant' do
+    result = subject.call(File.open(fixtures_dir + 'WEBP/unrecognised-variant.webp', 'rb'))
+    check_result(result, false)
+  end
+
+  it 'successfully parses lossy (VP8) WebP files' do
+    result = subject.call(File.open(fixtures_dir + 'WEBP/lossy.webp', 'rb'))
+    check_result(result, true, 181, 65)
+  end
+
+  it 'successfully parses lossless WebP files' do
+    result = subject.call(File.open(fixtures_dir + 'WEBP/lossless.webp', 'rb'))
+    check_result(result, true, 181, 65)
+  end
+
+  it 'successfully parses lossless WebP files with an alpha channel' do
+    result = subject.call(File.open(fixtures_dir + 'WEBP/lossless-alpha.webp', 'rb'))
+    check_result(result, true, 181, 65, false, true)
+  end
+
+  it 'successfully parses extended WebP files' do
+    result = subject.call(File.open(fixtures_dir + 'WEBP/extended.webp', 'rb'))
+    check_result(result, true, 181, 65)
+  end
+
+  it 'successfully parses extended WebP files with an alpha channel' do
+    result = subject.call(File.open(fixtures_dir + 'WEBP/extended-alpha.webp', 'rb'))
+    check_result(result, true, 181, 65, false, true)
+  end
+
+  it 'successfully parses extended WebP files with animation' do
+    result = subject.call(File.open(fixtures_dir + 'WEBP/extended-animation.webp', 'rb'))
+    check_result(result, true, 211, 211, true, true)
+  end
+
+  private
+
+  def check_result(
+    result,
```

Comment:
not sure about DRYing up the code this much. I understand that most of the things being checked by these tests are fairly the same, thus you are trying to avoid repetition. But the moment for example you start adding extra stuff, like an EXIF, and you'll want to test it, you'll need to add an extra argument in here as "expected behaviour".. and then one more and one more again, risking to make the method declaration quite long.

Plus it requires you going back and forward to see "oh what did I pass to this method? (scroll up) Ah yes I passed TRUE (scroll down). And what was the height_px? (scroll up).. etc" . I'd rather have a test well compacted in one place and that's it. Helpers method re-used can be extracted of course and declared separately, but this is not one of those. This is actual logic being checked.

Reply:
What Enrico said ☝️ I support the idea of being more declarative and less _smart_ when it comes to writing tests, as it improves readability, which is arguably the most important aspect with tests.

Reply:
Fair point. I do have a tendency to lean towards DRYness in situations like this, probably a holdover from working in payments code where there are lots of lovely little edge cases where almost everything else remains the same 😅. It'll hurt my heart a bit, but good shout, I'll switch it up 👍🏻 .